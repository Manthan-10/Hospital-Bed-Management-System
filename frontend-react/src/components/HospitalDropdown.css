
.hsp-dropdown {
  background-color:white;
  padding: 20px;
  /* margin-left:auto;
  margin-right:15px;
  margin-top: 10px; */

}

.select-container {
  background-color:gray;
}

body{
    margin:0;
    background-color: rgb(199, 206, 206);
}