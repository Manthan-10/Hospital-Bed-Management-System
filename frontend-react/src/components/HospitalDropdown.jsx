import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import { fetchPatientIdSuccess } from "../Redux/PatientSlice";
import BedService from "../services/bed.service";

import "./HospitalDropdown.css";

/**
 * This component displays all the Hospitals in a dropdown
 *
 */
const HospitalDropdown = () => {
  const [data, setHospitalList] = useState([]);
  const { t } = useTranslation();

  /**
   * @return Details of all the Hospitals
   *
   */
  useEffect(() => {
    BedService.getHospitals()
      .then((response) => {
        console.log(response);
        setHospitalList(response.data);
      })
      .catch((e) => {
        console.log(e);
      });
  }, []);

  let navigate = useNavigate();
  const dispatch = useDispatch();

  /**
   * Handles Choosing of Hospitals in a dropdown
   */
  function handleChange(value) {
    if (value === "1") {
      navigate(`/hospitals/${value}/floor/1`);
      dispatch(fetchPatientIdSuccess(null));
      window.location.reload(false);
    } else {
      navigate(`/hospitals/${value}/floor/4`);
      dispatch(fetchPatientIdSuccess(null));
      window.location.reload(false);
    }
  }

  return (
    <div className="hsp-dropdown">
      <select
        className="select-container"
        size="large"
        onChange={(event) => handleChange(event.target.value)}
      >
        <option>{t("Select_Hospital")}</option>
        {data.map((hospital) => (
          <option key={hospital.id} value={hospital.id}>
            {hospital.name}
          </option>
        ))}
      </select>
    </div>
  );
};

export default HospitalDropdown;
