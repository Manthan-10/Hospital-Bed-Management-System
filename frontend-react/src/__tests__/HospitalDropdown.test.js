import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import HospitalDropdown from "../components/HospitalDropdown";
import Adapter from "enzyme-adapter-react-16";
import * as reactRouterDom from "react-router-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

Enzyme.configure({ adapter: new Adapter() });

/**
 * Testing for UI of Login page
 * @author MN097789
 */

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useSelector: jest.fn(),

  useNavigate: () => mockNavigate,
}));
const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");

describe("HospitalDropdown UI Test Suite", () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store;

  test("Checks if it renders correctly", () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <HospitalDropdown />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
