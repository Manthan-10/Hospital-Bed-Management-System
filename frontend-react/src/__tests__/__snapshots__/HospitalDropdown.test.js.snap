// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HospitalDropdown UI Test Suite Checks if it renders correctly 1`] = `
<div
  className="hsp-dropdown"
>
  <select
    className="select-container"
    onChange={[Function]}
    size="large"
  >
    <option>
      Select_Hospital
    </option>
  </select>
</div>
`;
